cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -1062,7 +1075,16 @@ private int moveLeadershipInBatch() {
         // Mark leadership movements in progress.
         _executionTaskManager.markTasksInProgress(leadershipMovementTasks);
 
-        // Run preferred leader election.
+        // Run preferred leader election unless there is already an ongoing leadership movement.
```

Comment:
nit: can we make this chunk of code before line 1075. It grantees that we do not do anything until PLE zNode is cleared.